Artifacts/MsSql/Structure/Security/0040-Tables-MultipleAuthStrats.sql

```diff
@@ -0,0 +1,131 @@
+ON DELETE CASCADE
+GO
+
+-- FOREIGN KEY for [[ResourceClaimId]]  column
+ALTER TABLE [dbo].[ResourceClaimActions]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ResourceClaimActions_dbo.ResourceClaims_ResourceClaimId] FOREIGN KEY([ResourceClaimId])
+REFERENCES [dbo].[ResourceClaims] ([ResourceClaimId])
+ON DELETE CASCADE
+GO
+
+---------------------------- [dbo].[ClaimSetResourceClaimActions] --------------------------------------------------------
+
+CREATE TABLE [dbo].[ClaimSetResourceClaimActions](
+	[ClaimSetResourceClaimActionId] [int] IDENTITY(1,1) NOT NULL,
+	[ActionId] [int] NOT NULL,
+	[ClaimSetId] [int] NOT NULL,
+	[ResourceClaimId] [int] NOT NULL,
+	[ValidationRuleSetNameOverride] [nvarchar](255) NULL,
+ CONSTRAINT [PK_dbo.ClaimSetResourceClaimActions] PRIMARY KEY CLUSTERED 
+(
+	[ClaimSetResourceClaimActionId] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
+) ON [PRIMARY]
+GO
+
+-- NONCLUSTERED INDEX for [ActionId],[ClaimSetId],[ResourceClaimId] columns 
+CREATE NONCLUSTERED INDEX [IX_ActionId_ClaimSetId_ResourceClaimId]
+    ON [dbo].[ClaimSetResourceClaimActions]([ActionId],[ClaimSetId],[ResourceClaimId] ASC);
+GO
+
+-- FOREIGN KEY for [ActionId]  column
+ALTER TABLE [dbo].[ClaimSetResourceClaimActions]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActions_dbo.Actions_ActionId] FOREIGN KEY([ActionId])
+REFERENCES [dbo].[Actions] ([ActionId])
+ON DELETE CASCADE
+GO
+
+-- FOREIGN KEY for [ClaimSetId]  column
+ALTER TABLE [dbo].[ClaimSetResourceClaimActions]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActions_dbo.ClaimSets_ClaimSetId] FOREIGN KEY([ClaimSetId])
+REFERENCES [dbo].[ClaimSets] ([ClaimSetId])
+ON DELETE CASCADE
+GO
+
+-- FOREIGN KEY for [ResourceClaimId] column
+ALTER TABLE [dbo].[ClaimSetResourceClaimActions]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActions_dbo.ResourceClaims_ResourceClaimId] FOREIGN KEY([ResourceClaimId])
+REFERENCES [dbo].[ResourceClaims] ([ResourceClaimId])
+GO
+
+---------------------------- [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides] --------------------------------------------------------
+
+CREATE TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides](
+	[ClaimSetResourceClaimActionAuthorizationStrategyOverrideId] [int] IDENTITY(1,1) NOT NULL,
+	[ClaimSetResourceClaimActionId] [int] NOT NULL,
+	[AuthorizationStrategyId] [int] NOT NULL,
+ CONSTRAINT [PK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides] PRIMARY KEY CLUSTERED 
+(
+	[ClaimSetResourceClaimActionAuthorizationStrategyOverrideId] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
+) ON [PRIMARY]
+GO
+
+-- NONCLUSTERED INDEX for [ClaimSetResourceClaimActionId],[AuthorizationStrategyId] columns 
+CREATE NONCLUSTERED INDEX [IX_ClaimSetResourceClaimActionId_AuthorizationStrategyId]
```

Comment:
This needs to be a _unique_ constraint/index.

Reply:
I created CREATE UNIQUE INDEX here